sklearn/ensemble/_gb.py

```diff
@@ -578,7 +583,13 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # The requirements of _decision_function (called in two lines
             # below) are more constrained than fit. It accepts only CSR
             # matrices.
-            X = check_array(X, dtype=DTYPE, order="C", accept_sparse="csr")
+            X = check_array(
+                X,
+                dtype=DTYPE,
+                order="C",
+                accept_sparse="csr",
+                force_all_finite=False,
```

Comment:
On `main`, the finite check is reductant since `X` is already checked at `_validate_data`. I would be happy to have just this diff be a separate PR. (It would also help reduce the diff in this PR)

Would you be okay with opening another PR with just this part?

Reply:
I've created a PR for this part #22159.